app/assets/stylesheets/views/article.scss

```diff
@@ -34,7 +34,7 @@
         width: 100%;
         aspect-ratio: auto 650 / 273;
         object-fit: scale-down;
-        border-radius: var(--radius) var(--radius) 0 0;
+        border-radius: var(--radius-auto) var(--radius) 0 0;
```

Comment:
You need to do it to both the vars on this line

```css
border-radius: var(--radius-auto) var(--radius-auto) 0 0;
```

Reply:
That is what I thought, but in my testing that resulted the border radius never showing on the image when it should. Setting --radius-auto for only the first variable was the only thing that worked for me.

![image](https://user-images.githubusercontent.com/13600696/180424368-9fe8da30-3059-4b91-b9af-f746015e0def.png)

![image](https://user-images.githubusercontent.com/13600696/180424436-7bc273dc-9b9b-4286-ab4a-9e9b20d730a4.png)

Reply:
I've had a look it's caused by `--radius-auto` being `value / 0.375rem` as opposed to just `value`.

In order to use it in our use case we'd have to do `value value 0 0 / 0.375rem 0.375rem 0 0` which the variable doesn't offer but as it happens what you've written works because it outputs `value / 0.375rem 0.375rem 0 0` will becomes.

```css
border-top-left-radius: 0.375rem 0.375rem;
border-top-right-radius: 0.375rem 0.375rem;
border-bottom-right-radius: 0.375rem 0px;
border-bottom-left-radius: 0.375rem 0px;
```

TIL: border radius' allow some interesting customisation.

Reply:
That makes a lot of sense now. I didn't even fully understand why what I did worked 😆 

Would it make sense to create another variable?
```css
--radius-auto-single: Max(0px, Min(var(--radius), calc((100vw - 4px - 100%) * 9999))) var(--radius);
```

Then do something like this?
```css
border-top-left-radius: var(--radius-auto-single);
border-top-right-radius: var(--radius-auto-single);
border-bottom-left-radius: 0;
border-bottom-right-radius: 0;
```

I think this is much clearer and easier understand than my original PR. Maybe better variable names?

Reply:
Maybe, though we may need @Ridhwana to weigh in on that.

I'd say single does make sense but probably shouldn't have the secondary value. If we just have one like this

```css
--radius: 0.375rem;
--radius-large: 0.75rem;
--radius-auto-single: Max(0px, Min(var(--radius), calc((100vw - 4px - 100%) * 9999)));
--radius-large-auto-single: Max(0px, Min(var(--radius), calc((100vw - 4px - 100%) * 9999)));
--radius-auto: var(--radius-auto-single) / var(--radius);
--radius-large-auto: var(--radius-large-auto-single) / var(--radius-large);
```

Then for our code we can do

```css
border-radius: var(--radius-auto-single) var(--radius-auto-single) 0 0;
```

Reply:
Ah, I thought the second variable was to give it an arc shape. I didn't run the math, but if it always results in a perfect circle, then it can probably be removed.

Reply:
Testing locally I can see everything looks great with your changes @Fish1! 

I think you both make great points about the readability of what's going on here, my only other suggestion would be we use:

```css
border-radius: var(--radius-auto);
border-bottom-left-radius: 0;
border-bottom-right-radius: 0;
```

which seems to work well, and is _maybe_ more understandable?

I don't have a strong feeling, so I'm going to go ahead and approve this PR for now 😄 Absolutely happy for it to be refactored in a follow up, if you think it's a boost for the readability!